tests/random_test.py

```diff
@@ -1107,6 +1107,20 @@ class A: pass
     with self.assertRaisesRegex(TypeError, 'unrecognized type .* PRNG'):
       jax.random.key(42, impl=A())
 
+  def test_keyarray_custom_vjp(self):
+    # Regression test for https://github.com/google/jax/issues/18442
+    @jax.custom_vjp
+    def f(_, state):
+      return state
+    def _f_fwd(_, state):
+      return state, None
+    def _f_bwd(_, state_bar):
+      return state_bar
+    f.defvjp(_f_fwd, _f_bwd)
```

Comment:
What's the behavior if `symbolic_zeros=True`?

Reply:
This results in `TypeError: Value CustomVJPPrimal(value=8.0, perturbed=False) with type <class 'jax._src.custom_derivatives.CustomVJPPrimal'> is not a valid JAX type` somewhere in the custom vjp internals.

Reply:
Ah, that's because the test returns it as-is from the forward. We'd have to write a slightly different test to exercise symbolic zeros mode, maybe where the forward function tree-maps `lambda x: x.value` over its current output.

The reason to test this mode is that we expect RNG keys to be symbolically zero.

Reply:
Test is added, thanks!